File: maven-leetcode/src/main/java/leetcode/Num1.java
package leetcode;

public class Num1 {
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            /*
            * 思路：这是个O(n*2)的算法，如果序列是有序的，还可以变成O(n)的算法
            * 所以本题可以变成O(nlog(n))的算法。现在的思路就是两重循环遍历，有序序列可遍历一次即可*/
            int [] res = new int [2];
            point:for (int i=0; i<nums.length-1; i++){
                for (int j = i+1; j<nums.length; j++) {
                    if (nums[i] + nums[j] == target) {
                        res[0] = i;
                        res[1] = j;
                        break point;
                    }
                }
            }
            return res;
        }
    }
}

File: learnPython/num1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        '''
        如何和java一样采用O(n^2)去实现会超时，所以采用hash方式存储出现过的数据
        '''
        m = {}
        for i,v in enumerate(nums):
            if target-v in m:
                return [m[target-v],i]
            m[v] = i
        return -1